R/derive_merged.R

```diff
@@ -173,136 +180,188 @@
+#'   "DM",    "02",       64, "YEARS",
+#'   "DM",    "03",       85, "YEARS"
+#' ) %>%
+#'   mutate(STUDYID = "AB42")
 #'
-#' # Merging all dm variables to vs
 #' derive_vars_merged(
 #'   vs,
 #'   dataset_add = select(dm, -DOMAIN),
 #'   by_vars = exprs(STUDYID, USUBJID)
 #' ) %>%
-#'   select(STUDYID, USUBJID, VSTESTCD, VISIT, VSSTRESN, AGE, AGEU)
-#'
+#'   select(USUBJID, VSTESTCD, VISIT, VSSTRESN, AGE, AGEU)
+#'
+#' @caption Merge only the first/last value (`order` and `mode`)
+#' @info Merge the last occurring weight for each subject to the demographics dataset.
+#'
+#' - To enable sorting by visit date `convert_dtc_to_dtm()` is used to convert
+#'   to a datetime, within the `order` argument.
+#' - Then the `mode` argument is set to `"last"` to ensure the last sorted value
+#'   is taken. Be cautious if `NA` values are possible in the `order` variables -
+#'   see [Sort Order](https://pharmaverse.github.io/admiral/articles/generic.html#sort_order).
+#' - The `filter_add` argument is used to restrict the vital signs records only
+#'   to weight assessments.
+#' @code
+#' derive_vars_merged(
+#'   dm,
+#'   dataset_add = vs,
+#'   by_vars = exprs(STUDYID, USUBJID),
+#'   order = exprs(convert_dtc_to_dtm(VSDTC)),
+#'   mode = "last",
+#'   new_vars = exprs(LSTWT = VSSTRESN),
+#'   filter_add = VSTESTCD == "WEIGHT"
+#' ) %>%
+#'   select(USUBJID, AGE, AGEU, LSTWT)
+#'
+#' @caption Handling duplicates (`check_type`)
+#' @info The source records are checked regarding duplicates with respect to the
+#'   by variables and the order specified.
+#'   By default, a warning is issued if any duplicates are found.
+#'   Note the results here with a new vital signs dataset containing a
+#'   duplicate last weight assessment date.
+#' @code [expected_cnds = "duplicate_records"]
+#' vs_dup <- tribble(
+#'   ~DOMAIN,  ~USUBJID, ~VSTESTCD, ~VISIT,      ~VSSTRESN, ~VSDTC,
+#'   "VS",     "01",     "WEIGHT",  "WEEK 2",        81.1, "2013-09-24",
+#'   "VS",     "01",     "WEIGHT",  "WEEK 4",        82.6, "2013-09-24"
+#' ) %>%
+#'   mutate(STUDYID = "AB42")
 #'
-#' # Merge last weight to adsl
-#' adsl <- tribble(
-#'   ~STUDYID,   ~USUBJID, ~AGE,   ~AGEU,
-#'   "PILOT01", "01-1302",   61, "YEARS",
-#'   "PILOT01", "17-1344",   64, "YEARS"
-#' )
+#' derive_vars_merged(
+#'   dm,
+#'   dataset_add = vs_dup,
+#'   by_vars = exprs(STUDYID, USUBJID),
+#'   order = exprs(convert_dtc_to_dtm(VSDTC)),
+#'   mode = "last",
+#'   new_vars = exprs(LSTWT = VSSTRESN),
+#'   filter_add = VSTESTCD == "WEIGHT"
+#' ) %>%
+#'   select(USUBJID, AGE, AGEU, LSTWT)
+#'
+#' @info For investigating the issue, the dataset of the duplicate source records
+#'   can be obtained by calling `get_duplicates_dataset()`:
+#' @code
+#' get_duplicates_dataset()
+#'
+#' @info Common options to solve the issue:
+#' - Restricting the source records by specifying/updating the `filter_add` argument.
+#' - Specifying additional variables for `order` - this is the most common approach
```

Comment:
I'd also put this bullet point first as it's the most common option.

```suggestion
#' - Specifying additional variables for `order` - this is the most common approach,
```

Reply:
sure, will update